### PR TITLE
[RHOL-611] Add takeWhile and dropWhile implementation in StreamT

### PR DESCRIPTION
**Pull request by AbnerZheng**
## Overview
Add takeWhile and dropWhile implementation in StreamT

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
https://rchain.atlassian.net/browse/RHOL-611

### Complete this checklist before you submit the PR
- [X] This PR contains no more than 200 lines of code, excluding test code.
- [X] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [X] You assigned one person to review this PR

### CODE REVIEW

**Comment by AbnerZheng** [comment]
@KentShikama PTAL.

File: shared/src/test/scala/coop/rchain/shared/StreamTSpec.scala [hunk 1]
```diff
@@ -58,6 +58,16 @@ class StreamTSpec extends FunSpec with Matchers with GeneratorDrivenPropertyChec
 
     }
 
+    it(
+      "should allow taking the longest prefix of this StreamT whose elements satisfy the predicate") {
```
Comment (KentShikama) [inline shared/src/test/scala/coop/rchain/shared/StreamTSpec.scala]: Did you run casper/test or did the formatter do this?

**Comment by KentShikama** [comment]
@ArturGajowy @birchmd @goral09 Can one of you Scala experts take a look?

File: shared/src/test/scala/coop/rchain/shared/StreamTSpec.scala (hunk 1 above)
Comment (AbnerZheng) [reply shared/src/test/scala/coop/rchain/shared/StreamTSpec.scala]: I run the shared/test, and the formatter did this.

**Comment by KentShikama** [comment]
Thanks!

**Pull request merged**